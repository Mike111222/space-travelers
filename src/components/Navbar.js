import React from 'react';
import { Link } from 'react-router-dom';
import logo from '../img/logo.png';

const Navbar = () => (
  <nav className="navBar">
    <div className="navItems">
      <ul className="navLinks">
        <li><Link to="/">MyProfile</Link></li>
        <li><Link to="missions">Missions</Link></li>
        <li><Link to="rockets">Rockets</Link></li>
      </ul>
    </div>
    <div className="logoTitleContainer">
      <h2 className="navTitle">Space Travelers Hub</h2>
      <img className="logo" src={logo} alt="logo" />
    </div>
  </nav>
);

export default Navbar;
