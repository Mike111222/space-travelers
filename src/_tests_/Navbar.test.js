import { MemoryRouter } from 'react-router-dom';
import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Navbar from '../components/Navbar';

afterEach(() => {
  cleanup();
});

describe('Display Navbar', () => {
  it('Navbar renders correctly', () => {
    const tree = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
