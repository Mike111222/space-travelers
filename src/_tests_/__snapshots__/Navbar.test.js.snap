// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Navbar Navbar renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="navBar"
      >
        <div
          class="navItems"
        >
          <ul
            class="navLinks"
          >
            <li>
              <a
                href="/profile"
              >
                MyProfile
              </a>
            </li>
            <li>
              <a
                href="/missions"
              >
                Missions
              </a>
            </li>
            <li>
              <a
                href="/rockets"
              >
                Rockets
              </a>
            </li>
          </ul>
        </div>
        <div
          class="logoTitleContainer"
        >
          <h2
            class="navTitle"
          >
            Space Travelers Hub
          </h2>
          <img
            alt="logo"
            class="logo"
            src="logo.png"
          />
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="navBar"
    >
      <div
        class="navItems"
      >
        <ul
          class="navLinks"
        >
          <li>
            <a
              href="/profile"
            >
              MyProfile
            </a>
          </li>
          <li>
            <a
              href="/missions"
            >
              Missions
            </a>
          </li>
          <li>
            <a
              href="/rockets"
            >
              Rockets
            </a>
          </li>
        </ul>
      </div>
      <div
        class="logoTitleContainer"
      >
        <h2
          class="navTitle"
        >
          Space Travelers Hub
        </h2>
        <img
          alt="logo"
          class="logo"
          src="logo.png"
        />
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
